### ISSUE TITLE
Add `Tabs` accessibility primitive

### ISSUE THREAD

**Issue by joshpensky** [marathontvapp/carnation#1]
Adds a new `Tabs` accessibility primitive with the following API:

```tsx
<Tabs.Root value={value} onValueChange={onValueChange}>
  <Tabs.List>
    <Tabs.Trigger value="seasons" />
    <Tabs.Trigger value="episodes" />
  </Tabs.List>
  <Tabs.Content value="seasons">{/* ... */}</Tabs.Content>
  <Tabs.Content value="episodes">{/* ... */}</Tabs.Content>
</Tabs.Root>
```

## Acceptance Criteria
- [ ] Primitive API matches the above
- [ ] Primitive follows accessibility guidelines from APG: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/
- [ ] All sub-components of `Root` should be hot-swappable with custom components via `asChild`
